l5kit/l5kit/rasterization/semantic_rasterizer.py

```diff
@@ -162,47 +116,55 @@ def render_semantic_map(
+            lane_idx = self.mapAPI.bounds_info["lanes"]["ids"][lane_idx]
 
-            # Note(lberg): this called on all polygons skips some of them, don't know why
-            cv2.fillPoly(img, [lanes_area], (17, 17, 31), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
+            # interpolate over polyline to always have the same number of points
+            lane_coords = self.mapAPI.get_lane_as_interpolation(
+                lane_idx, INTERPOLATION_POINTS, InterpolationMethod.INTER_ENSURE_LEN
+            )
+            lanes_area[idx * 2] = lane_coords["xyz_left"][:, :2]
+            lanes_area[idx * 2 + 1] = lane_coords["xyz_right"][::-1, :2]
 
             lane_type = "default"  # no traffic light face is controlling this lane
-            lane_tl_ids = set([MapAPI.id_as_str(la_tc) for la_tc in lane.traffic_controls])
+            lane_tl_ids = set(self.mapAPI.get_lane_traffic_control_ids(lane_idx))
             for tl_id in lane_tl_ids.intersection(active_tl_ids):
-                if self.proto_API.is_traffic_face_colour(tl_id, "red"):
+                if self.mapAPI.is_traffic_face_colour(tl_id, "red"):
                     lane_type = "red"
-                elif self.proto_API.is_traffic_face_colour(tl_id, "green"):
+                elif self.mapAPI.is_traffic_face_colour(tl_id, "green"):
                     lane_type = "green"
-                elif self.proto_API.is_traffic_face_colour(tl_id, "yellow"):
+                elif self.mapAPI.is_traffic_face_colour(tl_id, "yellow"):
                     lane_type = "yellow"
+            lanes_mask[lane_type][idx * 2 : idx * 2 + 2] = True
+
+        if len(lanes_area):
+            lanes_area = cv2_subpixel(transform_points(lanes_area.reshape((-1, 2)), raster_from_world))
 
-            lanes_lines[lane_type].extend([xy_left, xy_right])
+            for lane_area in lanes_area.reshape((-1, INTERPOLATION_POINTS * 2, 2)):
+                # need to for-loop otherwise some of them are empty
+                cv2.fillPoly(img, [lane_area], (17, 17, 31), **CV2_SUB_VALUES)
 
-        cv2.polylines(img, lanes_lines["default"], False, (255, 217, 82), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
-        cv2.polylines(img, lanes_lines["green"], False, (0, 255, 0), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
-        cv2.polylines(img, lanes_lines["yellow"], False, (255, 255, 0), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
-        cv2.polylines(img, lanes_lines["red"], False, (255, 0, 0), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
+            lanes_area = lanes_area.reshape((-1, INTERPOLATION_POINTS, 2))
+            cv2.polylines(img, lanes_area[lanes_mask["default"]], False, (255, 217, 82), **CV2_SUB_VALUES)
+            cv2.polylines(img, lanes_area[lanes_mask["red"]], False, (255, 0, 0), **CV2_SUB_VALUES)
+            cv2.polylines(img, lanes_area[lanes_mask["yellow"]], False, (255, 255, 0), **CV2_SUB_VALUES)
+            cv2.polylines(img, lanes_area[lanes_mask["green"]], False, (0, 255, 0), **CV2_SUB_VALUES)
```

Comment:
I would define a constant for the colours `(255, 217, 82), (255, 0, 0)`,  etc, associated w/ the `default`, `red`, etc.

Reply:
+1